### PR TITLE
Lidar processing and progressive reward

### PR DESCRIPTION
**Pull request by Doge-God**
For auto encode experiments:
- Tools to train & setup 1D AE for testing
- Support for running with modified TD3AE (not pushed in CARES RL yet

For progressive goal:
- Added tools to generate parametric description for track center based on way point we're already using to place goals

Changes to CarTrackEnvironment:
- Fixed incorrect observation in step()
- Extracted method ways to modify observation & reward function quickly for experiments. Currently configurable via some constants at the top of the class def. Not using config with the rationale being some settings might be specific to the environment.

### CODE REVIEW

File: src/controllers/launch/sim_localized.launch.py
Comment (emilysteiner71) [inline src/controllers/launch/sim_localized.launch.py:1]: What is the discernible difference between this and sim.launch.py? The only differences I can find are:
- sim_localized never runs tracking (the same as running sim with tracking : False)
- sim_localized includes your extra nodes in the return statement but I can't see why they aren't also returned in sim.launch.py

File: src/controllers/launch/sim.launch.py [hunk 1]
```diff
@@ -88,13 +90,61 @@ def generate_launch_description():
...
 
     return LaunchDescription([
         #TODO: Find a way to remove this
         SetEnvironmentVariable(name='GZ_SIM_RESOURCE_PATH', value=pkg_f1tenth_description[:-19]),
         SetParameter(name='use_sim_time', value=True),
         environment,
         alg,
-        sim,
+        sim
         #algorithm
```
Comment (emilysteiner71) [inline src/controllers/launch/sim.launch.py:149]: Should you return the new nodes you've created?

File: austin_1_save.pgm
Comment (emilysteiner71) [inline austin_1_save.pgm:1]: Is this your logged version? If so, should not be added to repo and all logs should be saved in the logs folder (prevents them getting included in commits). If this is necessary for the code then it should be in the relevant package

File: lidar_ae_ftg_rand.pt
Comment (emilysteiner71) [inline lidar_ae_ftg_rand.pt:1]: same as austin_1_save.pgm. I can't view the contents of either file so not sure whether they serve a purpose or are just saved from running the program

File: austin_1_save.yaml
Comment (emilysteiner71) [inline austin_1_save.yaml:1]: yamls should be in the relevant config package. We want to minimize files floating around outside src if possible

File: src/environments/environments/F1tenthEnvironment.py
Comment (emilysteiner71) [inline src/environments/environments/F1tenthEnvironment.py:1]: Can't use these changes. Functions should always calculate steering angle primarily and only convert to twist when required, hence the conversion here

File: src/controllers/controllers/controller.py [hunk 2]
```diff
@@ -72,9 +98,14 @@ def __init__(self, node_name, car_name, step_length, lidar_points = 10):
 
         self.timer = self.create_timer(step_length, self.timer_cb)
         self.timer_future = Future()
+        
+        # Lidar processing 
+        self.ae_lidar_model = LidarConvAE()
+        # self.ae_lidar_model = BetaVAE1D(1,10,beta=4)
+        self.ae_lidar_model.load_state_dict(torch.load("/home/anyone/autonomous_f1tenth/lidar_ae_ftg_rand.pt"))
```
Comment (emilysteiner71) [inline src/controllers/controllers/controller.py:105]: This line breaks my program even with the file lidar_ae_ftg_rand.pt saved in the same location as pr'd. Regardless, breaking errors based on whether ae files are available should not be included (as it shouldn't impact trying to run traditional control with no aes). This could be fixed with a try except clause to only kill the program if you are trying to use ae and it doesn't load

**Review by emilysteiner71 — CHANGES_REQUESTED** [review]
Changes required are detailed in file comments :) looks good, just needs a few changes to prevent breaking other packages/programs

File: austin_1_save.yaml
Comment (Doge-God) [reply austin_1_save.yaml:1]: This and the pgm file are generated with SLAM and are needed for localization. I'm using the AMCL modules from Nav2 with the rationale being it should be faster than SLAM with less overhead since it is not continuously mapping. I'm moving them to a folder in environments dedicated to map infos now.

File: src/controllers/launch/sim_localized.launch.py
Comment (Doge-God) [reply src/controllers/launch/sim_localized.launch.py:1]: That was mainly there to test localization while using ftg to move the car. Getting rid of it now

Comment (Doge-God) [reply src/controllers/launch/sim_localized.launch.py:1]: I put the extra nodes in sim.launch.py since eventually it will be used by policies that need localization. Didnt put it in return since there was only ftg policy working and it didn't need these nodes.

**Review by emilysteiner71 — APPROVED** [review]
